### PR TITLE
fix(DB): Salma Saldean Missing Gossip

### PR DESCRIPTION
**Pull request by ZhengPeiRu21**
## Changes Proposed:
-  Add missing gossip for Salma Saldean (NPC 235)

## Issues Addressed:

- Closes unreported issue

## SOURCE:

Source: https://wowpedia.fandom.com/wiki/Salma_Saldean

## Tests Performed:

Capture

## How to Test the Changes:

1. .go creature 62837
2. See gossip is appear

## Known Issues and TODO List:

Please kindly feel free to change the Gossip Menu ID and NPC Text ID to anything else using variables at top of SQL file! I am just use these now as they are available. My chosen ID do not conflict with any of my other PR.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by acidmanifesto** [comment]
Gossip ids and gossips are sniffed. Do we have a verified database from where this matches up or a sniff to reference the id and text?

**Comment by ZhengPeiRu21** [comment]
> Gossip ids and gossips are sniffed. Do we have a verified database from where this matches up or a sniff to reference the id and text?

I would very appreciate if anyone can find correct IDs. As for if gossip is correct, I only know this is correct with Vmangos and Wiki, but do not have sniff, so verify this would also be good!

@acidmanifesto I have many similar other PR for gossip fix I am working on. Should I wait to create these until we have sniffers for verify, or should I create them so sniffers can know what to check?

**Comment by acidmanifesto** [comment]
Well vmangos is a acceptible source. Can u see if the gossip matches up with that?

File: data/sql/updates/pending_db_world/rev_1643833630496479400.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1643833630496479400');
+
+SET @NPC_ENTRY := 235; /* Salma Saldean */
+SET @GOSSIP_MENU_ID := 61027;
+SET @NPC_TEXT_ID := 50033;
+
+UPDATE `creature_template` SET `gossip_menu_id` = @GOSSIP_MENU_ID WHERE `entry` = @NPC_ENTRY;
+UPDATE `creature_template` SET `npcflag` = 3 WHERE `entry` = @NPC_ENTRY;
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1643833630496479400.sql:8]:
```suggestion
UPDATE `creature_template` SET `npcflag` = `npcflag`|1|2 WHERE `entry` = @NPC_ENTRY;
```
https://www.azerothcore.org/wiki/sql-standards#flags-amp-bits

Comment (ZhengPeiRu21) [reply data/sql/updates/pending_db_world/rev_1643833630496479400.sql]: Thank you! Sorry to have missed this SQL standard. I will fix for my future PRs.

**Comment by ZhengPeiRu21** [comment]
> Well vmangos is a acceptible source. Can u see if the gossip matches up with that?

https://github.com/vmangos/core/blob/a01720f547f0d042b7989085a1a3558cc6e92a64/sql/old_migrations/20170805164518_world.sql

Here is this gossip in Vmangos repo - the best I can find is in this migrations file because current data is not in raw SQL. 

But here is screenshot of it in Vmangos database running locally:
Capture

**Comment by acidmanifesto** [comment]
> > Well vmangos is a acceptible source. Can u see if the gossip matches up with that?
> 
> https://github.com/vmangos/core/blob/a01720f547f0d042b7989085a1a3558cc6e92a64/sql/old_migrations/20170805164518_world.sql
> 
> Here is this gossip in Vmangos repo - the best I can find is in this migrations file because current data is not in raw SQL. 
> 
> But here is screenshot of it in Vmangos database running locally:
> Capture
> 
> 

Well then. You jave propperly referenced and verified the information. I will approve the review. Great work.

**Review by acidmanifesto — APPROVED** [review]

**Comment by Kitzunu** [comment]
Thanks for the PR! But I have to say no to this specific way of doing this.

In this PR https://github.com/azerothcore/azerothcore-wotlk/pull/10526 I am implementing `quest_greetings` which will be the proper way to show gossips for creatures with only one option (quests)

Please update this PR accordingly.

**Pull request closed without merging**